util/build_configs/cray-internal/generate-modulefile.bash

```diff
@@ -112,29 +145,45 @@ if { ! [ info exists env(PE_ENV) ] } {
     module load PrgEnv-gnu
 }
 
-##
-## Now, we're going to attempt to get to the right build type
-###
-#
-set network seastar
-if { [ info exists env(XTPE_NETWORK_TARGET) ] } {
-    set network $env(XTPE_NETWORK_TARGET)
-} elseif { [ info exists env(CRAYPE_NETWORK_TARGET) ] } {
-    set network $env(CRAYPE_NETWORK_TARGET)
-}
-set accel 0
-if { [ info exists env(CRAY_ACCEL_TARGET) ] } {
-    set accel 1
-}
+set compiler $env(PE_ENV)
 
+if { [string match cray-shasta $CHPL_HOST_PLATFORM] } {
+    # Interim settings for Shasta systems.
 
-setenv MPICH_GNI_DYNAMIC_CONN disabled
-if { [string match gemini $network] } {
-    set CHPL_HOST_PLATFORM cray-xe
-} elseif { [string match aries $network] } {
-    set CHPL_HOST_PLATFORM cray-xc
-} else {
-    set CHPL_HOST_PLATFORM cray-xt
+    # So far we only have gnu-based Chapel for Shasta.
+    if { [string equal -nocase cray $compiler] } {
+        module swap PrgEnv-cray PrgEnv-gnu
+    } elseif { [string equal -nocase intel $compiler] } {
+        module swap PrgEnv-intel PrgEnv-gnu
+    } elseif { [string equal -nocase pgi $compiler] } {
+        module swap PrgEnv-pgi PrgEnv-gnu
+    }
+
+    # The cray-libsci module is loading hugepages, which we don't want (yet).
+    module unload cray-libsci
+
+    # We have to use the slurm-srun launcher.
+    setenv CHPL_LAUNCHER slurm-srun
```

Comment:
I'm surprised we're not auto-detecting this, any idea why we're not?

Reply:
Will fix.